### PR TITLE
Pr from fork/1038

### PR DESCRIPTION
**Pull request by rhornung67**
# Summary

- This PR pulls the branch from a PR submitted on a forked repo into the RAJA repo for CI testing and merging when approved. The original PR is: https://github.com/LLNL/RAJA/pull/1038

### CODE REVIEW

File: test/old-tests/unit/test-kernel.cpp
Comment (MrBurmark) [inline test/old-tests/unit/test-kernel.cpp:2051]: Shouldn't the extra 1 be at the end with the reordering?

Comment (rhornung67) [reply test/old-tests/unit/test-kernel.cpp:2051]: Good catch. I missed that.

Comment (corbett5) [reply test/old-tests/unit/test-kernel.cpp:2051]: Thanks, I missed this too, I just fixed this in the fork PR.

**Review by artv3 — DISMISSED** [review]

File: test/old-tests/unit/test-kernel.cpp
Comment (rhornung67) [reply test/old-tests/unit/test-kernel.cpp]: I fixed in here. Please double-check my change and approve if you think this is good to go. @corbett5 @MrBurmark is there an analogous change that should be made to the corresponding HIP policy?

Comment (corbett5) [reply test/old-tests/unit/test-kernel.cpp]: Thanks. I'm pretty sure there is. And also for flat `cuda_exec` it would be nice, but my use case at the moment involves kernel.

https://github.com/LLNL/RAJA/blob/dffd98b1ca3f4b0f8308fc0a78d1d66252250d47/include/RAJA/policy/hip/kernel/HipKernel.hpp#L193

Comment (rhornung67) [reply test/old-tests/unit/test-kernel.cpp]: @corbett5 we will fill in the gaps soon, may not be in time for the next RAJA release, which is imminent. Also, I restarted the GItlab CI on corona. Not sure why it timed out without starting the builds.

**Comment by corbett5** [comment]
Is there anything else that I should do? I see that the LC CI fails, but that appears to be a HIP CMake error that I don't believe is related to this PR.

```
-- HIP Support is ON
CMake Error at /usr/tce/packages/cmake/cmake-3.14.5/share/cmake/Modules/FindPackageHandleStandardArgs.cmake:137 (message):
  Could NOT find HIP (missing: HIP_HIPCC_EXECUTABLE HIP_HIPCONFIG_EXECUTABLE
  HIP_PLATFORM HIP_COMPILER HIP_RUNTIME)
Call Stack (most recent call first):
  /usr/tce/packages/cmake/cmake-3.14.5/share/cmake/Modules/FindPackageHandleStandardArgs.cmake:378 (_FPHSA_FAILURE_MESSAGE)
  blt/cmake/thirdparty/FindHIP.cmake:183 (find_package_handle_standard_args)
  blt/cmake/thirdparty/SetupHIP.cmake:13 (find_package)
  blt/cmake/thirdparty/SetupThirdParty.cmake:56 (include)
  blt/SetupBLT.cmake:112 (include)
  CMakeLists.txt:153 (include)
-- Configuring incomplete, errors occurred!
```

**Comment by rhornung67** [comment]
> Is there anything else that I should do? I see that the LC CI fails, but that appears to be a HIP CMake error that I don't believe is related to this PR.
> 
> ```
> -- HIP Support is ON
> CMake Error at /usr/tce/packages/cmake/cmake-3.14.5/share/cmake/Modules/FindPackageHandleStandardArgs.cmake:137 (message):
>   Could NOT find HIP (missing: HIP_HIPCC_EXECUTABLE HIP_HIPCONFIG_EXECUTABLE
>   HIP_PLATFORM HIP_COMPILER HIP_RUNTIME)
> Call Stack (most recent call first):
>   /usr/tce/packages/cmake/cmake-3.14.5/share/cmake/Modules/FindPackageHandleStandardArgs.cmake:378 (_FPHSA_FAILURE_MESSAGE)
>   blt/cmake/thirdparty/FindHIP.cmake:183 (find_package_handle_standard_args)
>   blt/cmake/thirdparty/SetupHIP.cmake:13 (find_package)
>   blt/cmake/thirdparty/SetupThirdParty.cmake:56 (include)
>   blt/SetupBLT.cmake:112 (include)
>   CMakeLists.txt:153 (include)
> -- Configuring incomplete, errors occurred!
> ```

@corbett5 the issue is that LC removed the version of HIP we were using in our CI when they upgraded corona. We;ve fixed this and I'm updating the PR branches as PRs get merged to not overuse CZ resources for CI. It'll run and pass at some point soon.

**Comment by rhornung67** [comment]
@corbett5 The CI is passing. I can't approve this since I made this PR from the one in your forked repo. 

@MrBurmark @ajkunen @artv3 can you review this?  This is only a piece. I made a GIthHub issue to propagate this change to all the other places in the code where it makes sense. But, I don't want to hold up the release until that's done.

**Review by artv3 — APPROVED** [review]
Looks good to me. Imagine there is probably something similar for HIP right? We should make a note to add that variant later.

**Review by MrBurmark — APPROVED** [review]

**Pull request merged**
